src/client/pages/Registration.tsx

```diff
@@ -20,14 +19,16 @@ import useRecaptcha, {
 import { DetailedRecaptchaError } from '@/client/components/DetailedRecaptchaError';
 import locations from '@/client/lib/locations';
 import { EmailInput } from '@/client/components/EmailInput';
+import { buildUrlWithQueryParams } from '@/shared/lib/routeUtils';
+import { QueryParams } from '@/shared/model/QueryParams';
 import { GeoLocation } from '@/shared/model/Geolocation';
 
 export type RegistrationProps = {
   returnUrl?: string;
   email?: string;
   recaptchaSiteKey: string;
   oauthBaseUrl: string;
-  queryString?: string;
+  queryString: QueryParams;
```

Comment:
not sure if the name of the prop makes sense anymore as it's not a string we're passing in now, it's an object, in this file and other places this is being used

my suggestion would be something like

```suggestion
  queryParams: QueryParams;
```